tle/util/cache_system2.py

```diff
@@ -239,6 +239,13 @@ def keep(problem):
         self.problem_by_name = problem_by_name
         self.problems_last_cache = time.time()
 
+        for problem in self.problems:
+            problem_contest: cf.Contest = self.cache_master.contest_cache.contest_by_id.get(problem.contestId)
```

Comment:
I see why u use type hint but it does not really match the coding style used everywhere else. i would not start using it at single places.